### PR TITLE
provide torch device to totalspineseg

### PR DESCRIPTION
**Pull request by NathanMolinier**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/onboarding/software-development.html#pr-labels) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Testing) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

The aim of this PR is to make totalspineseg usable with GPU

## Linked issues

https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4721

### CODE REVIEW

**Comment by joshuacwnewton** [comment]
Tagging @naga-karthik as the author of https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4721

**Review by joshuacwnewton — APPROVED** [review]
Since the change is so simple/straightforward, I'm approving (in terms of the code change/syntax).

But, I don't have a GPU computer to test with, so I will wait until someone else can test to merge this change.

**Comment by NathanMolinier** [comment]
I tested it and it worked on our internal server

**Pull request merged**
